modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/searchable/SearchableMenu.ts

```diff
@@ -0,0 +1,167 @@
+      // again. We can't just use the originalSandbox, because that will
+      // have been thrown away and recreated by now.
+      const newSandboxComp = Coupling.getCoupled(dropdown, 'sandbox');
+      optSearcherState.each(
+        (searcherState) => findWithinSandbox(newSandboxComp).each(
+          (inputComp) => restoreState(inputComp, searcherState)
+        )
+      );
+    }
+  );
+};
+
+// This event is triggered by the searcher for key events
+// that should be handled by the currently selected item
+// (that is, the one with *fake* focus, not real focus). So we
+// need to redispatch them to the selected item in the sandbox.
+const handleRedirectToMenuItem = (sandboxComp: AlloyComponent, se: SimulatedEvent<RedirectMenuItemInteractionEvent>): void => {
+  getActiveMenuItemFrom(sandboxComp).each(
+    (activeItem) => {
+
+      AlloyTriggers.retargetAndDispatchWith(
+        sandboxComp,
+        activeItem.element,
+        se.event.eventType,
+        se.event.interactionEvent
+      );
+    }
+  );
+};
+
+// This function is useful when you have fakeFocus, so you can't just find the
+// currently focused item (or the item that triggered a key event). It relies on
+// the following relationships between components
+// The Sandbox creates a tieredmenu, so Sandboxing.getState returns the TieredMenu
+// The TieredMenu uses Highlighting for managing which menus are active, so
+// Highlighting.getHighlighted(tmenu) is the current active menu
+// The Menu uses highlighting to manage the active item, so use
+// Highlighting.getHighlighted(menu) to get the current item.
+const getActiveMenuItemFrom = (sandboxComp: AlloyComponent): Optional<AlloyComponent> => {
+  // Consider moving some of these things into shared APIs. For example, make an extra API
+  // for TieredMenu to get the highlighted item.
+  return Sandboxing.getState(sandboxComp).bind(
+    Highlighting.getHighlighted
+  ).bind(
+    Highlighting.getHighlighted
+  );
+};
+
+const getSearchResults = (activeMenuComp: AlloyComponent): Optional<SugarElement<Element>> => {
+  // Depending on the menu layout, the search results will either be the entire
+  // menu, or something within the menu.
+  return Class.has(activeMenuComp.element, searchResultsClass)
+    ? Optional.some(activeMenuComp.element)
+    : SelectorFind.descendant(activeMenuComp.element, '.' + searchResultsClass);
+};
+
+// Model the interaction with ARIA
+const updateAriaOnHighlight = (tmenuComp: AlloyComponent, menuComp: AlloyComponent, itemComp: AlloyComponent): void => {
+  // This ARIA behaviour is based on the algolia example documented in TINY-8952
+  findWithinMenu(tmenuComp).each((inputComp) => {
+    setActiveDescendant(inputComp, itemComp);
+
+    const optActiveResults = getSearchResults(menuComp);
+    optActiveResults.each(
+      (resultsElem) => {
+        // Link aria-controls of the input to the id of the results container.
+        const controlledId = Attribute.get(resultsElem, 'id');
```

Comment:
Likewise here, should this be using `Attribute.getOpt` instead of doing an undefined check?

Reply:
- [ ] use Attribute.getOpt